### ISSUE TITLE
The system cannot find the path specified.

### ISSUE THREAD

**Issue by singhabhilash** [microsoft/WSL#7212]
### Windows Build Number

Microsoft Windows [Version 10.0.19043.1147]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

4.19.128 

### Distro Version

Ubuntu 20.04

### Repro Steps

Not sure how to reproduce this, it was working fine a few days back. Most probably some windows update broke it. I've gone through similar threads in the past, those didn't help. I've two distros. The first one I was using from long time. Trying to start it results in - 
 ```The system cannot find the path specified.```

Tried installing a new distro (Ubuntu 20.04) from the store. That fails to start as well with - 

WslRegisterDistribution failed with error: 
Error: 0x80070003 The system cannot find the path specified.

Tried disabling and enabling wsl feature as well.

### Expected Behavior

Expected the distro to start.

### Actual Behavior

Have mentioned the errors I'm seeing above. Can provide more info as necessary. Thanks.

**Comment by singhabhilash** [comment]
Tried uninstalling past few days of windows updates as well, still observing the same issue.

**Comment by benhillis** [comment]
@singhabhilash - do you happen to have a custom kernel defined in %userprofile%\.wslconfig?

**Comment by singhabhilash** [comment]
Hi, thanks for your time. I don't have .wslconfig file at all in that directory. Attaching the logs that I collected reading up the readme - 
[wsl.zip](https://github.com/microsoft/WSL/files/6856971/wsl.zip)

**Comment by benhillis** [comment]
@singhabhilash - can you double check that `%windir%\system32\lxss\tools\kernel` and `%windir%\system32\lxss\tools\initrd.img` are present?

**Comment by singhabhilash** [comment]
Yep, both files are there.

**Comment by singhabhilash** [comment]
Hi @benhillis , anything else I should try here? Thanks.

**Comment by singhabhilash** [comment]
Tried a couple of things but in vain. Surprisingly, I was able to get a new distro running by changing back to wsl 1. I wanted to continue with wsl 2 though.

With no other options left, I decided to upgrade to Windows 11(there were some other threads where people just suggested reinstalling windows!), what could go wrong? Turns out upgrading to Windows 11(on an unsupported 6th gen pc with some registry hacks), and voila the distro is working again! Not recommended if you don't have a fallback pc!!
